### DISCUSSION TITLE
N-to-N service dependencies

### DISCUSSION THREAD

**Discussion by shykes** [dagger/dagger#12156] [Ideas]
## Problem

The current services API on allows 1-to-1, one-way service dependencies. The following topologies are not allowed:

1. Service A needs to connect to B, *and vice-versa*
2. Service A, B, C all need to connect to each other. Similar to how Docker-compose networks work out of the box.

## Solution

Expand the Dagger API to support a wider variety of network topologies, without losing the benefits of the current design.

**Comment by marcosnils** [comment]
:wave:  adding an update here to share that #8641 already allows this in a simplified way.
